modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/action/UpdateCertificateMVCActionCommand.java

```diff
@@ -104,6 +107,16 @@ protected void doProcessAction(
 			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		if (!permissionChecker.isCompanyAdmin()) {
```

Comment:
isn't this too much a restriction? are we sure we can just allow admins here?

Reply:
It is what you must have for the portlet itself https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/SamlAdminPortlet.java#L66

Reply:
ok... so same question then... is it ok for us to restrict only for admin in that way?

other question, do we need to update this? https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/SamlAdminPortlet.java#L50

Reply:
The role-ref doesn't need all those roles because we're not referencing them all I think . But it has no effect because they always exist in the portal anyway.
Maybe we should discuss with the team if it would be worth while introducing resource actions for the SAML modules in general?

Reply:
I thought those were the roles allowed to execute the portlet... :exploding_head:

Reply:
No they are JSR-286 standard roles that get mapped to portal roles. If portal roles don't exist they get auto created when the portlet deploys. Which is important to enable admins to assign the role to users so the portlet can work when it checks for them later in code.